lib/fable/event.ex

```diff
@@ -43,15 +43,33 @@ defmodule Fable.Event do
   end
 
   defp event_ecto_spec(module) do
-    {:embed,
-     %Ecto.Embedded{
+    struct = %Ecto.Embedded{
        cardinality: :one,
        field: :data,
        on_cast: nil,
        on_replace: :raise,
        owner: __MODULE__,
        related: module,
        unique: true
-     }}
+     }
+     
+    if dependency_vsn_match?(:ecto, "~> 3.5") do
```

Comment:
This could be accomplished at compile time right?

Reply:
Ah, yeah. I mostly made sure it wasn't blowing up on my side :D